File: front/rupu-central/src/services/modules/horizontal-properties/parking/infrastructure/repositories/response/parking.response.ts
/**
 * Interfaces para las respuestas del backend de parqueaderos
 */

export interface BackendParkingZone {
  id: number;
  business_id: number;
  name: string;
  code: string;
  description?: string;
  location?: string;
  total_slots: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface BackendParkingZoneList {
  id: number;
  name: string;
  code: string;
  location: string;
  total_slots: number;
  is_active: boolean;
}

export interface BackendParkingZonesPaginatedResponse {
  success: boolean;
  data: BackendParkingZoneList[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface BackendParkingZoneResponse {
  success: boolean;
  message: string;
  data: BackendParkingZone;
}

export interface BackendParkingSlot {
  id: number;
  parking_zone_id: number;
  parking_type_id: number;
  slot_number: string;
  is_covered: boolean;
  width?: number;
  length?: number;
  height?: number;
  has_charger: boolean;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface BackendParkingSlotList {
  id: number;
  parking_zone_id: number;
  parking_zone_name: string;
  parking_type_id: number;
  parking_type_name: string;
  slot_number: string;
  is_covered: boolean;
  has_charger: boolean;
  is_active: boolean;
  is_available: boolean;
  is_assigned: boolean;
  is_occupied: boolean;
}

export interface BackendParkingSlotsPaginatedResponse {
  success: boolean;
  data: BackendParkingSlotList[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface BackendParkingSlotResponse {
  success: boolean;
  message: string;
  data: BackendParkingSlot;
}

export interface BackendParkingAssignment {
  id: number;
  business_id: number;
  parking_slot_id: number;
  property_unit_id?: number;
  resident_id?: number;
  vehicle_plate: string;
  vehicle_brand?: string;
  vehicle_model?: string;
  vehicle_color?: string;
  start_date: string;
  end_date?: string;
  is_active: boolean;
  notes?: string;
  created_at: string;
  updated_at: string;
}

export interface BackendParkingAssignmentList {
  id: number;
  parking_slot_id: number;
  parking_slot_number: string;
  property_unit_id?: number;
  property_unit_number: string;
  resident_id?: number;
  resident_name: string;
  vehicle_plate: string;
  vehicle_brand?: string;
  vehicle_model?: string;
  start_date: string;
  end_date?: string;
  is_active: boolean;
}

export interface BackendParkingAssignmentsPaginatedResponse {
  success: boolean;
  data: BackendParkingAssignmentList[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface BackendParkingAssignmentResponse {
  success: boolean;
  message: string;
  data: BackendParkingAssignment;
}

export interface BackendParkingReservation {
  id: number;
  business_id: number;
  parking_slot_id: number;
  property_unit_id?: number;
  resident_id?: number;
  visitor_id?: number;
  visitor_vehicle_id?: number;
  reservation_status_id: number;
  reservation_date: string;
  start_time: string;
  end_time: string;
  duration_hours: number;
  vehicle_plate: string;
  vehicle_brand?: string;
  vehicle_model?: string;
  vehicle_color?: string;
  qr_code?: string;
  access_code?: string;
  checked_in_at?: string;
  checked_out_at?: string;
  requires_approval: boolean;
  notes?: string;
  resident_notes?: string;
  created_at: string;
  updated_at: string;
}

export interface BackendParkingReservationList {
  id: number;
  parking_slot_id: number;
  parking_slot_number: string;
  property_unit_id?: number;
  property_unit_number: string;
  resident_id?: number;
  resident_name: string;
  visitor_id?: number;
  visitor_name: string;
  status_name: string;
  reservation_date: string;
  start_time: string;
  end_time: string;
  vehicle_plate: string;
  checked_in_at?: string;
  checked_out_at?: string;
  created_at: string;
}

export interface BackendParkingReservationsPaginatedResponse {
  success: boolean;
  data: BackendParkingReservationList[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface BackendParkingReservationResponse {
  success: boolean;
  message: string;
  data: BackendParkingReservation;
}

export interface BackendCheckParkingAvailabilityResponse {
  success: boolean;
  available: boolean;
  message?: string;
}

File: back/central-reserve/services/horizontalproperty/parking/internal/infra/primary/handlers/response/parking_response.go
package response

import "time"

// ParkingZoneResponse - Response de zona de parqueo
type ParkingZoneResponse struct {
	ID          uint      `json:"id"`
	BusinessID  uint      `json:"business_id"`
	Name        string    `json:"name"`
	Code        string    `json:"code"`
	Description string    `json:"description"`
	Location    string    `json:"location"`
	TotalSlots  int       `json:"total_slots"`
	IsActive    bool      `json:"is_active"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

// ParkingZoneListResponse - Response de lista de zonas de parqueo
type ParkingZoneListResponse struct {
	ID         uint   `json:"id"`
	Name       string `json:"name"`
	Code       string `json:"code"`
	Location   string `json:"location"`
	TotalSlots int    `json:"total_slots"`
	IsActive   bool   `json:"is_active"`
}

// PaginatedParkingZonesResponse - Response paginado de zonas de parqueo
type PaginatedParkingZonesResponse struct {
	Success    bool                      `json:"success"`
	Data       []ParkingZoneListResponse `json:"data"`
	Total      int64                     `json:"total"`
	Page       int                       `json:"page"`
	PageSize   int                       `json:"page_size"`
	TotalPages int                       `json:"total_pages"`
}

// ParkingSlotResponse - Response de espacio de parqueo
type ParkingSlotResponse struct {
	ID            uint      `json:"id"`
	ParkingZoneID uint      `json:"parking_zone_id"`
	ParkingTypeID uint      `json:"parking_type_id"`
	SlotNumber    string    `json:"slot_number"`
	IsCovered     bool      `json:"is_covered"`
	Width         *float64  `json:"width"`
	Length        *float64  `json:"length"`
	Height        *float64  `json:"height"`
	HasCharger    bool      `json:"has_charger"`
	IsActive      bool      `json:"is_active"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// ParkingSlotListResponse - Response de lista de espacios de parqueo
type ParkingSlotListResponse struct {
	ID              uint   `json:"id"`
	ParkingZoneID   uint   `json:"parking_zone_id"`
	ParkingZoneName string `json:"parking_zone_name"`
	ParkingTypeID   uint   `json:"parking_type_id"`
	ParkingTypeName string `json:"parking_type_name"`
	SlotNumber      string `json:"slot_number"`
	IsCovered       bool   `json:"is_covered"`
	HasCharger      bool   `json:"has_charger"`
	IsActive        bool   `json:"is_active"`
	IsAvailable     bool   `json:"is_available"`
	IsAssigned      bool   `json:"is_assigned"`
	IsOccupied      bool   `json:"is_occupied"`
}

// PaginatedParkingSlotsResponse - Response paginado de espacios de parqueo
type PaginatedParkingSlotsResponse struct {
	Success    bool                      `json:"success"`
	Data       []ParkingSlotListResponse `json:"data"`
	Total      int64                     `json:"total"`
	Page       int                       `json:"page"`
	PageSize   int                       `json:"page_size"`
	TotalPages int                       `json:"total_pages"`
}

// ParkingAssignmentResponse - Response de asignación
type ParkingAssignmentResponse struct {
	ID             uint       `json:"id"`
	BusinessID     uint       `json:"business_id"`
	ParkingSlotID  uint       `json:"parking_slot_id"`
	PropertyUnitID *uint      `json:"property_unit_id"`
	ResidentID     *uint      `json:"resident_id"`
	VehiclePlate   string     `json:"vehicle_plate"`
	VehicleBrand   string     `json:"vehicle_brand"`
	VehicleModel   string     `json:"vehicle_model"`
	VehicleColor   string     `json:"vehicle_color"`
	StartDate      time.Time  `json:"start_date"`
	EndDate        *time.Time `json:"end_date"`
	IsActive       bool       `json:"is_active"`
	Notes          string     `json:"notes"`
	CreatedAt      time.Time  `json:"created_at"`
	UpdatedAt      time.Time  `json:"updated_at"`
}

// ParkingAssignmentListResponse - Response de lista de asignaciones
type ParkingAssignmentListResponse struct {
	ID                 uint       `json:"id"`
	ParkingSlotID      uint       `json:"parking_slot_id"`
	ParkingSlotNumber  string     `json:"parking_slot_number"`
	PropertyUnitID     *uint      `json:"property_unit_id"`
	PropertyUnitNumber string     `json:"property_unit_number"`
	ResidentID         *uint      `json:"resident_id"`
	ResidentName       string     `json:"resident_name"`
	VehiclePlate       string     `json:"vehicle_plate"`
	VehicleBrand       string     `json:"vehicle_brand"`
	VehicleModel       string     `json:"vehicle_model"`
	StartDate          time.Time  `json:"start_date"`
	EndDate            *time.Time `json:"end_date"`
	IsActive           bool       `json:"is_active"`
}

// PaginatedParkingAssignmentsResponse - Response paginado de asignaciones
type PaginatedParkingAssignmentsResponse struct {
	Success    bool                            `json:"success"`
	Data       []ParkingAssignmentListResponse `json:"data"`
	Total      int64                           `json:"total"`
	Page       int                             `json:"page"`
	PageSize   int                             `json:"page_size"`
	TotalPages int                             `json:"total_pages"`
}

// ParkingReservationResponse - Response de reserva
type ParkingReservationResponse struct {
	ID                  uint       `json:"id"`
	BusinessID          uint       `json:"business_id"`
	ParkingSlotID       uint       `json:"parking_slot_id"`
	PropertyUnitID      *uint      `json:"property_unit_id"`
	ResidentID          *uint      `json:"resident_id"`
	VisitorID           *uint      `json:"visitor_id"`
	ReservationStatusID uint       `json:"reservation_status_id"`
	ReservationDate     time.Time  `json:"reservation_date"`
	StartTime           string     `json:"start_time"`
	EndTime             string     `json:"end_time"`
	DurationHours       float64    `json:"duration_hours"`
	VehiclePlate        string     `json:"vehicle_plate"`
	QRCode              string     `json:"qr_code"`
	AccessCode          string     `json:"access_code"`
	CheckedInAt         *time.Time `json:"checked_in_at"`
	CheckedOutAt        *time.Time `json:"checked_out_at"`
	CreatedAt           time.Time  `json:"created_at"`
}

// ParkingReservationListResponse - Response de lista de reservas
type ParkingReservationListResponse struct {
	ID                 uint       `json:"id"`
	ParkingSlotID      uint       `json:"parking_slot_id"`
	ParkingSlotNumber  string     `json:"parking_slot_number"`
	PropertyUnitID     *uint      `json:"property_unit_id"`
	PropertyUnitNumber string     `json:"property_unit_number"`
	ResidentID         *uint      `json:"resident_id"`
	ResidentName       string     `json:"resident_name"`
	VisitorID          *uint      `json:"visitor_id"`
	VisitorName        string     `json:"visitor_name"`
	StatusName         string     `json:"status_name"`
	ReservationDate    time.Time  `json:"reservation_date"`
	StartTime          string     `json:"start_time"`
	EndTime            string     `json:"end_time"`
	VehiclePlate       string     `json:"vehicle_plate"`
	CheckedInAt        *time.Time `json:"checked_in_at"`
	CheckedOutAt       *time.Time `json:"checked_out_at"`
	CreatedAt          time.Time  `json:"created_at"`
}

// PaginatedParkingReservationsResponse - Response paginado de reservas
type PaginatedParkingReservationsResponse struct {
	Success    bool                             `json:"success"`
	Data       []ParkingReservationListResponse `json:"data"`
	Total      int64                            `json:"total"`
	Page       int                              `json:"page"`
	PageSize   int                              `json:"page_size"`
	TotalPages int                              `json:"total_pages"`
}

// CheckParkingAvailabilityResponse - Response de verificación de disponibilidad
type CheckParkingAvailabilityResponse struct {
	Success   bool   `json:"success"`
	Available bool   `json:"available"`
	Message   string `json:"message,omitempty"`
}

// SuccessResponse - Response genérico de éxito
type SuccessResponse struct {
	Success bool        `json:"success"`
	Message string      `json:"message"`
	Data    interface{} `json:"data,omitempty"`
}

// ErrorResponse - Response de error
type ErrorResponse struct {
	Success bool   `json:"success"`
	Message string `json:"message"`
	Error   string `json:"error,omitempty"`
}

// ParkingTypeResponse - Response de tipo de parqueadero
type ParkingTypeResponse struct {
	ID          uint      `json:"id"`
	Name        string    `json:"name"`
	Code        string    `json:"code"`
	Description string    `json:"description"`
	IsActive    bool      `json:"is_active"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

// ParkingTypesResponse - Response de lista de tipos de parqueadero
type ParkingTypesResponse struct {
	Success bool                  `json:"success"`
	Data    []ParkingTypeResponse `json:"data"`
}
